### PR TITLE
fix: add nonce of transaction to execute to label

### LINKED ISSUE #655: [Queued list] the text above queued about tx in the next section is missing
## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: rinkeby

## Steps to reproduce
 1. Go to https://web-core.pages.dev/rin:0xf2565317F3Ae8Ae9EA98E9Fe1e7FADC77F823cbD/transactions/queue
 Current result:
image

## Expected result
QUEUE - TRANSACTION WITH NONCE 303 NEEDS TO BE EXECUTED FIRST

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #655 

## How this PR fixes it

" - transaction with nonce {{nonce}} needs to be executed first" has been added to "Queued" labels.

## How to test it

Queue two transactions. Observe the extra text on the second `LABEL`.

## Screenshots

image

### CODE REVIEW

**Review by katspaugh — APPROVED** [review]
👍

File: src/components/transactions/GroupLabel/index.tsx [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+import { isMultisigExecutionInfo, isTransactionListItem } from '@/utils/transaction-guards'
 
 const GroupLabel = ({ item }: { item: Label }): ReactElement => {
-  return <div className={css.container}>{item.label}</div>
+  let label: string | LabelValue = item.label
+
+  const { page } = useTxQueue()
+
+  if (label === LabelValue.Queued && !page?.next) {
+    const firstTx = page?.results.find(isTransactionListItem)
```
Comment (katspaugh) [inline src/components/transactions/GroupLabel/index.tsx:13]: Maybe we should display the `safeNonce` instead?

Comment (katspaugh) [reply src/components/transactions/GroupLabel/index.tsx:13]: Also how is `!page?.next` related to this label?

Comment (iamacook) [reply src/components/transactions/GroupLabel/index.tsx:13]: It's possible to set a transaction nonce higher than that of the Safe nonce though. This mirrors that of `safe-react` as well.

Comment (iamacook) [reply src/components/transactions/GroupLabel/index.tsx:13]: If there is `!page?.next`, there will be no 'Next' transaction label to append the nonce to.

Comment (iamacook) [reply src/components/transactions/GroupLabel/index.tsx:13]: In hindsight, that wouldn't be valid for long queues. I'm looking into this further.

File: src/components/transactions/GroupLabel/index.tsx [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
 
 const GroupLabel = ({ item }: { item: Label }): ReactElement => {
-  return <div className={css.container}>{item.label}</div>
+  let label: string | LabelValue = item.label
+
+  const { page } = useTxQueue()
+
+  const labels = page?.results.filter(isLabelListItem)
+
+  // Current page has both 'Queued' and 'Next' labels
```
Comment (katspaugh) [inline src/components/transactions/GroupLabel/index.tsx:14]: `First page` would be more correct.

File: src/components/transactions/GroupLabel/index.tsx [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
+  const hasBothLabels = labels && labels.length > 1
+
+  if (hasBothLabels && label === LabelValue.Queued) {
```
Comment (katspaugh) [inline src/components/transactions/GroupLabel/index.tsx:17]: Code style: I would move the bothLabels thing into the if like so:

```
// We're dealing with the "Queued" transactions that can be executed only after the "Next" transactions
if (label === LabelValue.Queued) {
  const hasNext = page?.results.some((tx) => isLabelListItem(tx) && tx !== item)
  const firtstTx...
}
```

**Review by katspaugh — COMMENTED** [review]
Worth extracting into a hook and adding tests?

File: src/components/transactions/GroupLabel/index.tsx [hunk 4]
```diff
@@ -1,9 +1,47 @@
...
+
+  // There is also a 'Next' label on the page of the queue
+  if (page.results.some((tx) => isLabelListItem(tx) && tx !== item)) {
+    return getQueueLabel(firstTx.transaction.executionInfo.nonce)
+  }
+
+  // First transaction has an out of order nonce
+  if (firstTx.transaction.executionInfo.nonce !== safe.nonce) {
+    return getQueueLabel(safe.nonce)
+  }
```
Comment (katspaugh) [inline src/components/transactions/GroupLabel/index.tsx:36]: That's a bit hard to read.
Suggestion:
```
const hasNextLabel = page.results.some((tx) => isLabelListItem(tx) && tx !== item)
const nonceInFuture = firstTx.transaction.executionInfo.nonce !== safe.nonce

if (!hasNextLabel && !nonceInFuture) return label

const nextNonce = nonceInFuture ? safe.nonce : firstTx.transaction.executionInfo.nonce

return `${label} - transaction with nonce ${nextNonce} needs to be executed first`
```

**Review by katspaugh — APPROVED** [review]
✅

**Pull request merged**